src/generator/pollers.ts

```diff
@@ -53,50 +56,60 @@ type ${poller.name} struct {
 	pt pollingTracker
 }
 
-// Done returns true if the polling operation has terminated either in a success case or failure case,
-// otherwise it will return false
-func (p *${poller.name}) Done() bool {
-	return p.pt.hasTerminated()
-}
-
-// ID ... 
-func (p *${poller.name}) ID() string {
-	return ""
+// Poll returns false if there was an error or polling has reached a terminal state
+func (p *${poller.name}) Poll(ctx context.Context) bool {
+	if done, err := p.done(ctx); err != nil {
+		return false
+	} else {
+		return !done
+	}
 }
 
-func (p *${poller.name}) Poll(ctx context.Context) (*${responseType}, error) {
-	if done, err := p.done(ctx); !done || err != nil {
-		return nil, err
-	}
+// Response returns the final response from the polling operation
+func (p *${poller.name}) Response() (*${responseType}, error) {
 	resp := p.response()
+	if resp == nil {
+		return nil, errors.New("did not find a response on the poller")
+	}
 	result, err := p.client.${poller.operationName}HandleResponse(resp)
 	if err != nil {
 		return nil, err
 	}
 	return result, nil
 }
 
+// ResumeToken generates the string token that can be used with the Resume${pascalCase(poller.name)} method
+// on the client to create a new poller from the data held in the current poller type
+func (p *${poller.name}) ResumeToken() (string, error) {
+	if p.pt.hasTerminated() {
+		return "", errors.New("cannot create a ResumeToken from a poller in a terminal state")
+	}
+	js, err := json.Marshal(p.pt)
+	if err != nil {
+		return "", fmt.Errorf("json.Marshal: %s", err.Error())
+	}
+	return string(js), nil
+}
+
 // Wait will continue to poll until a terminal state is reached or an error is encountered. Wait will use the 
 // duration specified in the retry-after header, if the header is not specified then the pollingInterval that
 // is specified will be used to wait between polling requests. 
 func (p *${poller.name}) Wait(ctx context.Context, pollingInterval time.Duration) (*${responseType}, error) {
-	for {
-		resp, err := p.Poll(context.Background())
-		if err != nil {
-			return nil, err
-		}
-		if p.Done() {
-			return resp, err
-		}
+	for p.Poll(context.Background()) {
 		if delay, found := p.response().RetryAfter(); found && delay > 0 {
```

Comment:
When will this be updated to use `azcore` v0.6.0 that just returns `int`?

Reply:
I'll update that in this PR, it slipped my mind.

Reply:
Done